examples/examples/error.rs

```diff
@@ -0,0 +1,68 @@
+#![deny(rust_2018_idioms)]
+use std::error::Error;
+use std::fmt;
+use tracing_error::{ErrorLayer, SpanTrace};
+use tracing_subscriber::{fmt::Layer as FmtLayer, prelude::*, registry::Registry};
+
+#[tracing::instrument]
+fn do_something(foo: &str) -> Result<&'static str, impl Error + Send + Sync + 'static> {
+    match do_another_thing(42, false) {
+        Ok(i) => Ok(i),
+        Err(e) => Err(FooError::new("something broke, lol", e)),
+    }
+}
+#[derive(Debug)]
+struct FooError {
+    message: &'static str,
+    source: Box<dyn Error + Send + Sync + 'static>,
+    context: SpanTrace,
```

Comment:
Nit: a comment explaining why a `SpanTrace` is needed would be helpful.

Reply:
(Note: I _know_ why you need it, but over-explaining a new concept can't hurt...)